inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_sequences_to_sequences_ie.cpp

```diff
@@ -23,14 +23,12 @@ ngraph::pass::ConvertLSTMSequenceMatcher::ConvertLSTMSequenceMatcher() {
             return false;
         }
 
-        const auto& W = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
-                lstm_sequence->input_value(4).get_node_shared_ptr());
+        const auto& W = lstm_sequence->input_value(4).get_node_shared_ptr();
         if (!W) {
             return false;
         }
 
-        const auto& R = std::dynamic_pointer_cast<ngraph::opset4::Constant>(
-                lstm_sequence->input_value(5).get_node_shared_ptr());
+        const auto& R = lstm_sequence->input_value(5).get_node_shared_ptr();
```

Comment:
If this nodes are used as inputs you need to avoid of get_node_shared_ptr call.

Reply:
fixed